crates/code_generation/src/cyclers.rs

```diff
@@ -115,14 +115,21 @@ fn generate_struct(cycler: &Cycler, cyclers: &Cyclers, mode: Execution) -> Token
     } else {
         Default::default()
     };
+    let own_subscribed_outputs_reader = if mode == Execution::ImageExtraction {
+        Default::default()
+    } else {
+        quote! {
+            own_subscribed_outputs_reader: framework::Reader<std::collections::HashSet<String>>,
+        }
+    };
 
     quote! {
         pub(crate) struct Cycler<HardwareInterface>  {
             instance: CyclerInstance,
             hardware_interface: std::sync::Arc<HardwareInterface>,
             own_writer: framework::Writer<Database>,
             own_changed: std::sync::Arc<tokio::sync::Notify>,
-            own_subscribed_outputs_reader: framework::Reader<std::collections::HashSet<String>>,
+            #own_subscribed_outputs_reader
```

Comment:
You could also give it an orphan reader but keeping the argument. Should we change the cycler API for a medium important tool? Same for the `Execution` enum extension. Would it be possible to implement the imagine tool with zero changes in the code generation of the cyclers?

Reply:
I now reimplemented the whole image extractor to extend the replayer struct instead. This will also spin up a fully fledged communication server, but results only in code generation changes of the replayer struct. This also allowed me to revert the extension of the `Execution` enum.

Reply:
Thanks!